### PR TITLE
feat: Start Schematic stack with Docker Compose

### LINKED ISSUE #1609: [Task] Start Schematic stack with Docker Compose
### What projects is this story for?

Schematic

### As a user, I want

As a developer, I want the components of the stack (services, DBs, etc.) to start in the correct order and when their dependencies are ready so that all the components can successfully start.

### Description

OpenChallenges has many components (services, DBs, etc.) that were started in the correct order using Nx `dependsOn` property (e.g. `openchallenges-challenge-service` depends on `openchallenges-mariadb`). The issue with this approach is that there was no guarantee that the dependencies would be fully started, which is often a requirement for a downstream component to start successfully. The solution implemented for OC was to add healthcheck to the Docker images and start the stack with Docker compose.

The goal of this ticket is to start the components of Schematic using the approach developed for OpenChallenges. While Schematic currently has a single component (`schematic-api`), this update will 1) make Schematic ready to add more components and start them properly and 2) use the same deployment method with Docker compose across all the stacks in the monorepo.

### Acceptance criteria

_No response_

### Tasks

_No response_

### Anything else?

_No response_

### Have you linked this story to a GitHub Project?

- [X] I have linked this story to a GitHub Project and set its metadata.

### PR DESCRIPTION
**Pull request by tschaffter**
Closes #1609

## Changelog

- Adopt the approach used by OpenChallenges to start the components of Schematic stack in the correct order

## Preview

Build all the images of the Schematic stack:

```console
schematic-build-images
```

Build a single image, e.g.:

```console
nx build-image schematic-api
```

Start a components and all its dependencies with Docker Compose:

```console
nx serve-detach schematic-api
```

### CODE REVIEW

File: docker/schematic/services/api.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+version: '3.8'
```
Comment (linglp) [inline docker/schematic/services/api.yml:1]: Is this version just a random number?

File: docker/schematic/serve-detach.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+
+args=(
+  # List of services in alphanumeric order
+  --file docker/schematic/services/api.yml
+
+  --file docker/schematic/networks.yml
+  --file docker/schematic/volumes.yml
+
```
[the remark this answers is not in the document]
Comment (tschaffter) [reply docker/schematic/serve-detach.sh:9]: I came up with this approach to break down long docker-compose.yml into smaller files that make it easier to identify the code that you may want to change.

[the remark this answers is not in the document]
Comment (linglp) [reply docker/schematic/serve-detach.sh:9]: This is interesting. Thank you for sharing.

**Review by linglp — APPROVED** [review]
Thanks @tschaffter . The PR looks good. I could spin up a schematic docker container using the instructions that you have above.

**Pull request merged**